src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs

```diff
@@ -239,14 +261,17 @@ public void SetExit()
         {
             // Can't exit from the top level of PSES
             // since if you do, you lose all LSP services
-            if (_psFrameStack.Count <= 1)
+            PowerShellContextFrame frame = CurrentFrame;
+            if ((frame.FrameType & PowerShellFrameType.Repl) is 0 || _psFrameStack.Count <= 1)
```

Comment:
Took me a minute but this was changed so that we set exit if we're on the last frame _or_ we're in a REPL frame, which needed to be added because we could have attached/remote frames around (screwing up the count)?

Reply:
Yup! Boy is that hard to read.

I think I'm going to add some props to `PowerShellContextFrame` like `bool IsRepl => etc` just to clean this up visually.